### PR TITLE
Examples: Replace deprecated "np.random.seed()" with "np.random.Generator()" (NPY002 warning by ruff)

### LINKED ISSUE #2779: ruff: Enable NPY (NumPy-specific) rules
Currently, ruff provides three NumPy-specific rules (https://docs.astral.sh/ruff/rules/#numpy-specific-rules-npy), which can help us avoid deprecated NumPy functions.

To enable NumPy-specific rules, we just need to add `NPY` to the `select` option below:

https://github.com/GenericMappingTools/pygmt/blob/8f1d476e0144fd51827a5a27d9fa387f622bb823/pyproject.toml#L91-L97 

After enabling the `NPY` rules, running `make format` gives many [NPY002]*https://docs.astral.sh/ruff/rules/numpy-legacy-random/) warnings, like:
```
examples/gallery/histograms/histogram.py:15:1: NPY002 Replace legacy `np.random.seed` call with `np.random.Generator`
   |
13 | import pygmt
14 |
15 | np.random.seed(100)
   | ^^^^^^^^^^^^^^ NPY002
16 |
17 | # Generate random elevation data from a normal distribution
   |
```
The usage of `np.random.seed(100)` is deprecated. We should use something like below instead:
```
rng = np.random.default_rng(100)
rng.random()
```

### PR DESCRIPTION
**Pull request by yvonnefroehlich**
**Description of proposed changes**

This PR aims to update the examples to follow ruff's NumPy rules, especially it addresses the `NPY002` warning (please see https://docs.astral.sh/ruff/rules/numpy-legacy-random/):
```
NPY002 Replace legacy `np.random.seed` call with `np.random.Generator`
```
Example:
```
import numpy as np
np.random.seed(42)
np.random.normal()
```
is replaced by
```
rng = np.random.default_rng(seed=42)
rng.normal()
```
Affected examples:
- [ ] gallery: histogram  - [old](https://www.pygmt.org/dev/gallery/histograms/histogram.html) -> [new](https://pygmt-dev--2781.org.readthedocs.build/en/2781/gallery/histograms/histogram.html)
- [ ] gallery: scatter_and_histograms - [old](https://www.pygmt.org/dev/gallery/histograms/scatter_and_histograms.html) -> [new](https://pygmt-dev--2781.org.readthedocs.build/en/2781/gallery/histograms/scatter_and_histograms.html)
- [ ] gallery: points - [old](https://www.pygmt.org/dev/gallery/symbols/points.html) -> [new](https://pygmt-dev--2781.org.readthedocs.build/en/2781/gallery/symbols/points.html)
- [ ] gallery: scatter - [old](https://www.pygmt.org/dev/gallery/symbols/scatter.html) -> [new](https://pygmt-dev--2781.org.readthedocs.build/en/2781/gallery/symbols/scatter.html)
- [ ] tutorial: cartesian_histograms - [old](https://www.pygmt.org/dev/tutorials/advanced/cartesian_histograms.html) -> [new](https://pygmt-dev--2781.org.readthedocs.build/en/2781/tutorials/advanced/cartesian_histograms.html)

Please note that the plots are not completely identical. This is because the generated random numbers are not identical.

Related to the PRs
- #2787
- #2782

Fixes #2779 partly

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If wrapping a new module, open a 'Wrap new GMT module' issue and submit reasonably-sized PRs.
- [ ] If adding new functionality, add an example to docstrings or tutorials.
- [ ] Use underscores (not hyphens) in names of Python files and directories.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

**Review by seisman — APPROVED** [review]

**Comment by seisman** [comment]
The PR title makes no sense to people who don't know NPY002. Could you please update the PR title to make it more clear what the changes are?

File: examples/gallery/histograms/scatter_and_histograms.py [hunk 1]
```diff
@@ -13,15 +13,16 @@
...
 
-np.random.seed(19680801)
-
 # Generate random data from a standard normal distribution centered on 0
-x = np.random.randn(1000)
-y = np.random.randn(1000)
+# with a standard deviation of 1
+rng = np.random.default_rng(seed=19680801)
+x = rng.normal(0, 1, 1000)
+y = rng.normal(0, 1, 1000)
```
Comment (weiji14) [inline examples/gallery/histograms/scatter_and_histograms.py:20]: Use keyword arguments to be clearer what the arguments mean?
```suggestion
y = rng.normal(loc=0, scale=1, size=1000)
x = rng.normal(loc=0, scale=1, size=1000)
```

File: examples/gallery/symbols/points.py [hunk 2]
```diff
@@ -11,10 +11,11 @@
 import pygmt
 
 # Generate a random set of points to plot
-np.random.seed(42)
+rng = np.random.default_rng(seed=42)
 region = [150, 240, -10, 60]
-x = np.random.uniform(region[0], region[1], 100)
-y = np.random.uniform(region[2], region[3], 100)
+x = rng.uniform(region[0], region[1], 100)
+y = rng.uniform(region[2], region[3], 100)
```
Comment (weiji14) [inline examples/gallery/symbols/points.py:17]:
```suggestion
x = rng.uniform(low=region[0], high=region[1], size=100)
y = rng.uniform(low=region[2], high=region[3], size=100)
```

File: examples/gallery/symbols/scatter.py [hunk 3]
```diff
@@ -13,18 +13,20 @@
...
+    region=[-1, 1, -1, 1],
     projection="X10c/10c",
-    frame=["xa0.2fg", "ya0.2fg", "WSrt"],
+    frame=["xa0.5fg", "ya0.5fg", "WSrt"],
 )
 for fill in ["gray73", "darkorange", "slateblue"]:
-    x, y = np.random.rand(2, n)  # random X and Y data in [0,1]
-    size = np.random.rand(n) * 0.5  # random size [0,0.5], in cm
+    x = rng.normal(0, 0.5, n)  # random x data
+    y = rng.normal(0, 0.5, n)  # random y data
```
Comment (weiji14) [inline examples/gallery/symbols/scatter.py:27]:
```suggestion
    x = rng.normal(loc=0, scale=0.5, size=n)  # random x data
    y = rng.normal(loc=0, scale=0.5, size=n)  # random y data
```

**Review by weiji14 — APPROVED** [review]
Minor suggestion to use kwargs, otherwise looks good! Just wait for the docs to be rebuilt and double-check that the transparency issue has been fixed with #2783.

**Comment by yvonnefroehlich** [comment]
> The PR title makes no sense to people who don't know NPY002. Could you please update the PR title to make it more clear what the changes are?

I have improved the title, added the links to the previews, and expanded the comment explaining this PR.

File: examples/gallery/histograms/scatter_and_histograms.py (hunk 1 above)
Comment (yvonnefroehlich) [reply examples/gallery/histograms/scatter_and_histograms.py]: I updated all examples to use the keyword arguments to have this consistent.

File: examples/gallery/symbols/scatter.py (hunk 3 above)
Comment (yvonnefroehlich) [reply examples/gallery/symbols/scatter.py]: I also updated this for `size` in the next line.

File: examples/gallery/symbols/scatter.py [hunk 4]
```diff
@@ -13,18 +13,22 @@
...
+    frame=["xa0.5fg", "ya0.5fg", "WSrt"],
 )
 for fill in ["gray73", "darkorange", "slateblue"]:
-    x, y = np.random.rand(2, n)  # random X and Y data in [0,1]
-    size = np.random.rand(n) * 0.5  # random size [0,0.5], in cm
+    # Generate standard normal distributions centered on 0
+    # with standard deviations of 1
+    x = rng.normal(loc=0, scale=0.5, size=n)  # random x data
+    y = rng.normal(loc=0, scale=0.5, size=n)  # random y data
+    size = rng.normal(loc=0, sclae=0.5, size=n) * 0.5  # random size, in cm
```
Comment (seisman) [inline examples/gallery/symbols/scatter.py:30]: typo
```suggestion
    size = rng.normal(loc=0, scale=0.5, size=n) * 0.5  # random size, in cm
```

Comment (yvonnefroehlich) [reply examples/gallery/symbols/scatter.py]: Oops, thanks!

File: examples/gallery/histograms/scatter_and_histograms.py (hunk 1 above)
Comment (yvonnefroehlich) [reply examples/gallery/histograms/scatter_and_histograms.py]: Thanks, no worries.

**Pull request merged**